holidays/countries/turks_and_caicos_islands.py

```diff
@@ -0,0 +1,126 @@
+        * [2023](https://www.facebook.com/photo/?fbid=557741379734203&set=a.363028765872133)
+        * [2025](https://www.facebook.com/photo/?fbid=1019082573584944&set=a.353200576839817)
+        * [Destination TCI](https://web.archive.org/web/20250429024708/https://destinationtci.tc/turks-and-caicos-islands-public-holidays/)
+        * [Time and Date](https://web.archive.org/web/20250429024853/https://www.timeanddate.com/holidays/turks-and-caicos-islands/)
+    """
+
+    country = "TC"
+    default_language = "en_TC"
+    # Separated from Jamaica in 1962.
+    start_year = 1963
+    supported_languages = ("en_TC", "en_US")
+    observed_label = "%s (observed)"
+
+    def __init__(self, *args, **kwargs):
+        ChristianHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
+        # Public Holidays Act 1980 established weekend-to-Monday rule.
+        kwargs.setdefault("observed_rule", SAT_SUN_TO_NEXT_MON)
+        kwargs.setdefault("observed_since", 1980)
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        # New Year's Day.
+        self._add_observed(self._add_new_years_day(tr("New Year's Day")))
+
+        # Commonwealth Day.
+        self._add_holiday_2nd_mon_of_mar(tr("Commonwealth Day"))
+
+        # Good Friday.
+        self._add_good_friday(tr("Good Friday"))
+
+        # Easter Monday.
+        self._add_easter_monday(tr("Easter Monday"))
+
+        self._add_holiday_last_mon_of_may(
+            # JAGS McCartney Day.
+            tr("JAGS McCartney Day")
+            if self._year >= 2020
+            # National Heroes Day.
+            else tr("National Heroes Day")
+        )
+
+        name = (
+            # King's Birthday.
+            tr("King's Birthday")
+            if self._year >= 2023
+            # Queen's Birthday.
+            else tr("Queen's Birthday")
+        )
+        dates_obs = {
+            2022: (JUN, 3),
+            2023: (JUN, 19),
+            2024: (JUN, 17),
+            2025: (JUN, 23),
+        }
+        if self._year in dates_obs:
+            self._add_holiday(name, dates_obs.get(self._year))
+        else:
+            self._add_holiday_2nd_mon_of_jun(name)
+
+        # Emancipation Day.
+        self._add_observed(self._add_holiday_aug_1(tr("Emancipation Day")))
+
+        # National Youth Day.
+        self._add_holiday_last_fri_of_sep(tr("National Youth Day"))
+
+        self._add_holiday_2nd_mon_of_oct(
+            # National Heritage Day.
+            tr("National Heritage Day")
+            if self._year >= 2014
+            # Columbus Day.
+            else tr("Columbus Day")
+        )
+
+        # National Day of Thanksgiving.
+        self._add_holiday_4th_fri_of_nov(tr("National Day of Thanksgiving"))
+
+        # Christmas Day.
+        self._add_observed(self._add_christmas_day(tr("Christmas Day")))
+
+        # Boxing Day.
+        self._add_observed(self._add_christmas_day_two(tr("Boxing Day")))
```

Comment:
```suggestion
        self._add_observed(
            # Christmas Day.
            self._add_christmas_day(tr("Christmas Day")),
            rule=SAT_SUN_TO_NEXT_MON_TUE,
        )

        self._add_observed(
            # Boxing Day.
            self._add_christmas_day_two(tr("Boxing Day")),
            rule=SAT_SUN_TO_NEXT_MON_TUE,
        )
```
As we can see in 2021, when December 26-26 falls on a Saturday-Sunday, non-working days are 27-28.